layerfetcher/layers.go

```diff
@@ -192,11 +193,24 @@ func (l *Logic) AddDBs(blockDB, AtxDB, TxDB, poetDB, IvDB, tbDB database.Getter)
 // layerHashReqReceiver returns the layer hash for the specified layer.
 func (l *Logic) layerHashReqReceiver(ctx context.Context, msg []byte) []byte {
 	lyr := types.NewLayerID(util.BytesToUint32(msg))
-	h := l.layerDB.GetLayerHash(lyr)
-	l.log.WithContext(ctx).With().Debug("responded layer hash request",
-		lyr,
-		log.String("layerHash", h.ShortString()))
-	return h.Bytes()
+	lyrHash := &layerHash{
+		ProcessedLayer: l.layerDB.ProcessedLayer(),
+		SimpleHash:     l.layerDB.GetLayerHash(lyr),
+		AggHash:        l.layerDB.GetAggregatedLayerHash(lyr),
+	}
+	out, err := types.InterfaceToBytes(lyrHash)
+	if err != nil {
+		l.log.WithContext(ctx).With().Error("failed to serialize layer hash",
```

Comment:
Panic is cleaner here. InterfaceToBytes returns error only if the input is incorrect (and in this case it is formed by the developer)

Reply:
done. thanks